vmdb/lib/workers/worker_base.rb

```diff
@@ -150,10 +151,15 @@ def log_prefix
   #
 
   def find_worker_record
-    @worker = self.class.corresponding_model.find_by_guid(@cfg[:guid])
+    @worker = self.class.corresponding_model.includes(:miq_server).find_by_guid(@cfg[:guid])
     do_exit("Unable to find instance for worker GUID [#{@cfg[:guid]}].", 1) if @worker.nil?
   end
 
+  def find_server_record
+    find_worker_record unless @worker
+    @server = @worker.miq_server
```

Comment:
Ah, there's a reason I didn't do a reader method, actually.

With the reader method, it would be a lazy initialization, and therefore still run the query from within the thread.  So, the query has to be run outside the thread and the ivar initialized before the thread is kicked off.

Unless there's some fanciness I'm not thinking about here.